adapters/repos/db/vector/hnsw/vector_cache.go

```diff
@@ -69,7 +69,7 @@ func (c *vectorCache) replaceMapIfFull() {
 	}
 }
 
-func (c *vectorCache) get(ctx context.Context, id int32) ([]float32, error) {
+func (c *vectorCache) get(ctx context.Context, id int64) ([]float32, error) {
```

Comment:
see above, all node ids should be represented by `uint64`, removing the need for the conversions.